### PR TITLE
feat(sns): support encryption of topics

### LINKED ISSUE #1729: [sns] support encrypted topics
TopicProps.builder() should have a withKmsMasterKeyId(String keyArn) method for setting the KmsMasterKeyId property. This would add support encrypted topics.

Currently the work around is to drop into the Resource and set the property but this was very hard to get to when first learning how to use the library. 
```
    public static void setKmsMasterKeyId(final Topic topic, final EncryptionKey encryptionKey){
        Resource cfnTopic = (Resource) topic.getNode().findChild("Resource");
        cfnTopic.addPropertyOverride("KmsMasterKeyId", encryptionKey.getKeyArn());
    }
```
Adding the property to TopicProps.builder() woule eliminate the need to drop into the Resource and make it easier to get off the ground for users new to the tool.

### PR DESCRIPTION
**Pull request by yamatatsu**
Add Support KMS masterKey on SNS.
It is allowed to set `masterKey` as `Kms::Key` to `TopicProps`.
Fixes #1729

### CODE REVIEW

**Comment by nija-at** [comment]
@yamatatsu 

> Is it possible for me to pass this test locally without a security token?

You do not need to run `npm run integ` just yet.  

`integ.sns.ts` synthesizes a CloudFormation template and compares it against the checked in template in [integ.sns.expected.json](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-sns/test/integ.sns.expected.json). The error you're seeing - `Error: Some stacks have changed.` - happens because you've updated the integration test but not the expected JSON.
Scrollback when you get this error and you will see the diff between what is expected and what is synthesized.

File: packages/@aws-cdk/aws-sns/test/test.sns.ts [hunk 1]
```diff
@@ -67,25 +68,36 @@ export = {
...
+      });
+
+      expect(stack).to(haveResource("AWS::SNS::Topic", {
+        "KmsMasterKeyId": { "Ref": "CustomKey1E6D0D07" },
+      }));
+
+      test.done();
+    },
+
     'specify both'(test: Test) {
```
[the remark this answers is not in the document]
Comment (yamatatsu) [reply packages/@aws-cdk/aws-sns/test/test.sns.ts]: OK. I'll revert it.
I just guessed that this test Intended to test all of `TopicProps` properties.
But I also think it is not needed to extend this test case.

**Comment by yamatatsu** [comment]
@nija-at 
I fixed the commit message. So it is needed to use `push -f`. Is it your expected? If not, I'll re-create PR.
And I fixed for your comments.

I have a question that I should edit `integ.sns.expected.json` manually? I guess that this json is a snapshot and is edited automatically.

**Comment by nija-at** [comment]
The team is currently heads down for the next couple of days, so it'll be a bit longer before we can get to this. Apologies for the delayed response on your PR.

**Comment by nija-at** [comment]
Code looks fine. We're not accepting community contributions into the CDK until next week, so hang tight.

**Comment by yamatatsu** [comment]
OK. It was so fun to complete making this PR with you.
ARIGATO!

**Pull request merged**